src/algokit/core/deploy.py

```diff
@@ -0,0 +1,157 @@
+    "ALGOD_PORT",
+    "INDEXER_TOKEN",
+    "INDEXER_SERVER",
+    "INDEXER_PORT",
+]
+
+DEPLOY_CONFIGS = {
+    "localnet": _generate_config(
+        algod_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        algod_server="http://localhost",
+        algod_port="4001",
+        indexer_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        indexer_server="http://localhost",
+        indexer_port="8980",
+    ),
+    "testnet": _generate_config(
+        algod_server="https://testnet-api.algonode.cloud",
+        indexer_server="https://testnet-idx.algonode.cloud",
+    ),
+    "mainnet": _generate_config(
+        algod_server="https://mainnet-api.algonode.cloud",
+        indexer_server="https://mainnet-idx.algonode.cloud",
+    ),
+    "betanet": _generate_config(
+        algod_server="https://betanet-api.algonode.cloud",
+        indexer_server="https://betanet-idx.algonode.cloud",
+    ),
+}
+
+
+def load_deploy_config(network: str, project_dir: Path) -> dict:
+    """
+    Parse the .env file if it exists and load configurations.
+    :param network: Network name.
+    :param project_dir: Project directory path.
+    :return: A dictionary containing the configuration.
+    """
+
+    env_file_path = project_dir / ".env"
+    if env_file_path.exists():
+        config = configparser.ConfigParser()
+        with env_file_path.open() as f:
+            config.read_string("[config]\n" + f.read())
+
+        # Extract network configuration
+        network_config = {}
+        prefix = f"{network.upper()}_"
+        for key, value in config["config"].items():
+            # Strip the network prefix from each key
+            if key.startswith(prefix):
+                new_key = key[len(prefix) :]
+                network_config[new_key] = value
+
+        # Fill missing keys with empty strings
+        for required_key in REQUIRED_KEYS:
+            if required_key not in network_config:
+                logger.warning(f"Warning: {required_key} is missing, filling with an empty string.")
+                network_config[required_key] = ""
+
+        return network_config
+    else:
+        logger.info(f"No .env file found, loading default configuration for {network}.")
+        return DEPLOY_CONFIGS.get(network, {})
+
+
+def load_deploy_command(network_name: str, project_dir: Path) -> str:
+    """
+    Load the deploy command for the given network from .algokit.toml file.
+    :param network_name: Network name.
+    :param project_dir: Project directory path.
+    :return: Deploy command.
+    """
+
+    # Load and parse the TOML configuration file
+    config_path = project_dir / ALGOKIT_CONFIG
+    try:
+        config_content = config_path.read_text("utf-8")
+    except FileNotFoundError:
+        logger.debug(f"No {ALGOKIT_CONFIG} file found in the project directory.")
+        raise click.ClickException(f"No {ALGOKIT_CONFIG} file found in the project directory.") from None
```

Comment:
why not raise from the original exception instead of `None` here?

Reply:
Done